tests/snippets/floats.py

```diff
@@ -229,3 +229,10 @@
 # Test float exponent:
 assert 1 if 1else 0 == 1
 
+pi = 3.14
+assert pi.__eq__(3.14)
+assert pi.__ne__(3.14) == False
+assert pi.__eq__(3) == False
+assert pi.__ne__(3)
+assert pi.__eq__('pi') == NotImplemented
+assert pi.__ne__('pi') == NotImplemented
```

Comment:
Because float is not fully ordered, please add `nan` and `inf` comparisons to ensure its completeness